zjit/src/hir.rs

```diff
@@ -2005,6 +2005,49 @@ impl Function {
             None => {},
         }
     }
+
+
+    /// Validates the following:
+    /// 1. Basic block jump args match parameter arity.
+    /// 2. Every terminator must be in the last position.
+    /// 3. Every block must have a terminator.
+    fn validate_block_terminators_and_jumps(&self) {
+        for block_id in self.rpo() {
+            let mut block_has_terminator = false;
+            let insns = &self.blocks[block_id.0].insns;
+            for (idx, insn_id) in insns.iter().enumerate() {
+                let insn = self.find(*insn_id);
+                if !insn.is_terminator() {
+                    continue;
+                }
+                if let Insn::Jump(BranchEdge{target, args}) = &insn {
```

Comment:
This should also check parameter matches for conditional jumps--IfFalse and IfTrue--which are not terminators